src/test/java/com/selfxdsd/selfweb/api/output/JsonProjectTestCase.java

```diff
@@ -0,0 +1,578 @@
+        Mockito.when(wallet.debt()).thenReturn(BigDecimal.valueOf(200));
+        Mockito.when(wallet.available()).thenReturn(BigDecimal.valueOf(800));
+        
+        Mockito.when(project.wallet()).thenReturn(wallet);       
+        
+        final JsonObject jsonProject = new JsonProject(project);
+        
+        MatcherAssert.assertThat(
+            jsonProject.getJsonObject("wallet"),
+            Matchers.equalTo(Json.createObjectBuilder()
+                .add("type", wallet.type())
+                .add("active", wallet.active())
+                .add("cash", wallet.cash().divide(BigDecimal.valueOf(100)))
+                .add("debt", wallet.debt().divide(BigDecimal.valueOf(100)))
+                .add(
+                    "available",
+                    wallet.available().divide(BigDecimal.valueOf(100))
+                ).build())
+        );       
+    }
+    
+    /**
+     * JsonProject has null repoFullName
+     */
+    
+    @Test
+    public void hasNullRepoFullName(){
+        final Project project = Mockito.mock(Project.class);
+        
+        Mockito.when(project.repoFullName()).thenReturn("null");
+        Mockito.when(project.provider()).thenReturn("Github");
+        
+        User owner = Mockito.mock(User.class);        
+        
+        Mockito.when(owner.username()).thenReturn("Andreea");        
+        
+        Mockito.when(project.owner()).thenReturn(owner);
+        
+        ProjectManager manager = Mockito.mock(ProjectManager.class);    
+        
+        Mockito.when(manager.id()).thenReturn(123);
+        Mockito.when(manager.userId()).thenReturn("321");
+        Mockito.when(manager.username()).thenReturn("charlsemike");
+        
+        Provider provider = Mockito.mock(Provider.class);
+        
+        Mockito.when(provider.name()).thenReturn("Github");
+        
+        Mockito.when(manager.provider()).thenReturn(provider);        
+        Mockito.when(manager.commission()).thenReturn(BigDecimal.valueOf(800));  
+        
+        Mockito.when(project.projectManager()).thenReturn(manager);
+        
+        Wallet wallet = Mockito.mock(Wallet.class);
+        
+        Mockito.when(wallet.type()).thenReturn("Stripe");
+        Mockito.when(wallet.active()).thenReturn(true);
+        Mockito.when(wallet.cash()).thenReturn(BigDecimal.valueOf(1000));
+        Mockito.when(wallet.debt()).thenReturn(BigDecimal.valueOf(200));
+        Mockito.when(wallet.available()).thenReturn(BigDecimal.valueOf(800));
+        
+        Mockito.when(project.wallet()).thenReturn(wallet);       
+        
+        final JsonObject jsonProject = new JsonProject(project);
+        
+        MatcherAssert.assertThat(
+            jsonProject.getString("repoFullName"),
+            Matchers.equalTo("null")
+        );   
+}
+    
+    /**
+     * JsonProject has null Provider
+     */
+    
+    
+    @Test
+    public void hasNullProvider(){
```

Comment:
@Lumi3011 This one can also be removed, it's logically identical to ``hasProvider()``